pandas/core/sparse/frame.py

```diff
@@ -73,6 +73,10 @@ def __init__(self, data=None, index=None, columns=None, default_kind=None,
             if columns is None:
                 raise Exception("cannot pass a series w/o a name or columns")
             data = {columns[0]: data}
+        elif isinstance(data, BlockManager):
+            fill_value_size = len(set(b.fill_value for b in data.blocks))
```

Comment:
@TomAugspurger @jreback 

Ok I need feedback on this change. Basically what this does is: if every SparseSeries's fill_value is the same it sets the default_fill_value to that. This to me seems really intuitive, but I trust your judgement. It also makes testing a whole of a hell lot easier which is why I did it.

I would argue pretty heavily that as a user if I added sparse series that all had fill_value=0 then I would expect the sparse data frame to have default_fill_value = 0 as well.

Reply:
yes this is the *only* reason to have a standalone ``SparseDataFrame``, meaning that they have the *same* default fill values. But its not worth the cost generally.